logger/redacting_formatter.go

```diff
@@ -0,0 +1,48 @@
+package logger
+
+import (
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+)
+
+type RedactionOption struct {
+	Field       string
+	Regexp      *regexp.Regexp
```

Comment:
Do we need a regexp here ?  Wouldn't a string be enough ?
I am afraid on performances with huge log volumes.

Reply:
It's a good question.
I tried to take an approach that would be configurable to each service.
If the Regexp is set to nil, the field will be redacted in it's entirety.

The reason the regexp is needed is for this issue: https://github.com/Scalingo/logs-service/issues/509 in which the path field (`path="/apps/66b24069fb0de6002981dd79/logs?timestamp=1727183062&token=[HERE]&stream=true"`) contains the token in the query string. In order to redact this, a regexp is required.

An alternative solution (perhaps better) is to avoid having the token in the query string in the first place, and move it to the request header instead, but that change is broader and has more impact.

What do you think is the better approach?

Reply:
Let's keep your approach and be vigilant at release !
The regexp you send from logs-service is compiled, this will help performance.